Makefile

```diff
@@ -131,9 +131,12 @@ perlcritic: tools/lib/
 .PHONY: test-unused-modules-changed
 test-unused-modules-changed:
 	@echo "[make] Unused modules check called over modified/new files only. For a full run use make test-unused-modules-full"
-	tools/detect_unused_modules -m `git --no-pager diff --name-only --diff-filter=d origin/master | grep '^tests/*' | grep -v '^tests/test_pods/'`
-	tools/detect_unused_modules -m `git --no-pager diff --unified=0 origin/master products/* | grep -oP "^-.*loadtest.*[\"']\K[^\"'].+(?=[\"'])"`
-	tools/detect_unused_modules -m `git --no-pager diff --unified=0 origin/master schedule/* | grep -oP "^-\s+- [\"']?\K.*(?=[\"']?)" | grep -v '{{'`
+	f=$$(mktemp)
+	git --no-pager diff --name-only --diff-filter=d origin/master | grep '^tests/*' | grep -v '^tests/test_pods/' > $$f
+	git --no-pager diff --unified=0 origin/master products/* | sed -n "s~^-.*loadtest\s\+\([\"']\)\([^\"']\+\)\1.*~tests/\2.pm~p" >> $$f
+	git --no-pager diff --unified=0 origin/master schedule/* | sed -n "s~^-\s\+-\s\+\([\"']\)\([^\"']\+\)\1.*~tests/\2.pm~p" | grep -v '{{' >> $$f
+	sort -u $$f -o $$f
+	tools/detect_unused_modules -m $$(cat $$f)
```

Comment:
In the latest step this became quite unreadable. Also I don't know if the temporary file is ensured to be pruned on failure. If you really need this then please put it into tools/detect_unused_modules instead of multi-level escaping in Makefile. Also your original "Add required quotes" is by far exceeded now. Please provide an according git commit message which explains the necessity